src/utils/filter-utils.js

```diff
@@ -369,6 +435,74 @@ export function filterData(dataset, filters) {
   return {data: filtered, filteredIndex, filteredIndexForDomain};
 }
 
+/**
+ * This returns the points to evaluate in polygon filter
+ * @param layer
+ * @param data
+ * @return {*[][]}
+ */
+export const getLayerPointsToFilter = (layer, data) => {
+  switch (layer.type) {
+    case LAYER_TYPES.arc:
+    case LAYER_TYPES.line:
+      const {
+        lng0,
+        lat0,
+        lng1,
+        lat1
+      } = layer.config.columns;
+      return [
+        [data[lng0.fieldIdx], data[lat0.fieldIdx]],
+        [data[lng1.fieldIdx], data[lat1.fieldIdx]]
+      ];
+    default:
+      const {lng, lat} = layer.config.columns;
+      return [
+        [data[lng.fieldIdx], data[lat.fieldIdx]]
+      ];
+  }
+};
+
+const filterDataMatchers = {
+  [FILTER_TYPES.range]: (data, filter, index, field) => {
+    const val = field ? data[field.tableFieldIndex - 1] : null;
+    return isInRange(val, filter.value)
+  },
+  [FILTER_TYPES.timeRange]: (data, filter, index, field) => {
+    const val = field ? data[field.tableFieldIndex - 1] : null;
+    const timeVal = field && field.filterProp && Array.isArray(field.filterProp.mappedValue)
+      ? field.filterProp.mappedValue[index]
+      : moment.utc(val).valueOf();
+
+    return isInRange(timeVal, filter.value);
+  },
+  [FILTER_TYPES.multiSelect]: (data, filter, index, field) => {
+    const val = field ? data[field.tableFieldIndex - 1] : null;
+    return filter.value.includes(val)
+  },
+  [FILTER_TYPES.select]: (data, filter, index, field) => {
+    const val = field ? data[field.tableFieldIndex - 1] : null;
+    return val === filter.value
+  },
+  // layers contain only layers for the current dataset
+  [FILTER_TYPES.polygon]: (data, filter, index, field, layers) => {
+    if (!(layers || layers.length === 0)) {
+      return true;
+    }
+
+    // determine which layers to apply the filter on
+    const currentLayers = filter.layerId.map(id =>
+        layers.find(l => l.id === id)
+      // we may get null value because filter.layerId may contain layers from other datasets
+    ).filter(l => Boolean(l));
+
+    return currentLayers.every(layer => {
+      const points = getLayerPointsToFilter(layer, data);
```

Comment:
Each layer has a method `getPositionAccessor` which will return a position accessor, you can get point position. no need for the `getLayerPointsToFilter` function
```
const getPosition = layer.getPositionAccessor();
const points = getPosition(data);
```